Feat/login controller

AoiTuNa opened the pull request:
이거 이번한번은 가능한가요... 일단 자바독 만들었고 로그인 관련한 js 폴더로 추가했습니다..
그래도 700개네요..

AoiTuNa committed:
login controller 준비 html 설정 준비

AoiTuNa committed:
간단한 로그인 로직 구현

AoiTuNa committed:
대강 기억나는 틀을 만듦

AoiTuNa committed:
login controller 준비 html 설정 준비

AoiTuNa committed:
login controller 구현 및 테스트 코드 작성

AoiTuNa committed:
login service 구현 및 테스트 코드 작성

AoiTuNa committed:
login service 구현 및 테스트 코드 작성

AoiTuNa committed:
로그인 로직 전체 테스트 코드 실행 성공 커버리지 86%

AoiTuNa committed:
참고용 Template 깃 푸쉬 제외

AoiTuNa committed:
javadoc 추가 작업

AoiTuNa committed:
안쓰는 import package 삭제

AoiTuNa committed:
안쓰는 import package 삭제

AoiTuNa committed:
안쓰는 import package 삭제

AoiTuNa committed:
필요한 파일만 넣어봄

AoiTuNa committed:
일단 돌아가게는 만듦

AoiTuNa commented:
필요하면 분리 할게요!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+            return "redirect:/pages/auth/login";
+        }
+        String accessToken = token.getAccessToken();
+        Cookie cookie = new Cookie("authorization",accessToken);
```
[P3] : accessToken 을 저장 할 때 쿠키의 key 값을 'authorization'으로 했을 시 리프레쉬 토큰 저장 시에 key 이름이 애매해질 것 같습니다

AoiTuNa replied:
refreshToken을 cookie에 저장하지 않고 유저 에이전트로 판단하는거 아니었나요..?

ckddms6530 replied:
무조건 사용한다는 건 아니고 그냥 이름이 애매하다 정도였습니다 ㅋㅋ

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    @PostMapping("/logout")
+    public String doLogout(HttpServletRequest req,RedirectAttributes redirectAttributes){
+        Cookie[] cookies = req.getCookies();
```
[P4] : 컨트롤러 메서드 파라미터로 쿠키 값을 직접 가져올 수 있습니다 밑에 링크 참고 해보시면 좋을 것 같습니다!
https://docs.spring.io/spring-framework/reference/web/webflux/controller/ann-methods/cookievalue.html

AoiTuNa replied:
추가해 봤습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+                    .map(Cookie::getValue)
+                    .orElse(null);
+            if(token == null){
+                throw new HttpClientErrorException(HttpStatus.FORBIDDEN);
```
[P3] : 액세스 토큰이 없는 사용자가 로그아웃 요청을 보낸 건 403 보다는 401 상태코드가 적절할 것 같습니다

AoiTuNa replied:
생각해보니 401이 적절해 보입니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+                    .orElse(null);
+            if(token == null){
+                throw new HttpClientErrorException(HttpStatus.FORBIDDEN);
+            }else{
```
[P4] 불필요한 else 문 같습니다 지우면 조금 더 깔끔한 코드가 될 것 같습니다!

AoiTuNa replied:
else문을 제거해도 문제가 없음을 확인했습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/domain/JwtToken.java:
```diff
@@ -0,0 +1,19 @@
+ * @version 1.0
+ */
+@Getter
+@Setter
```
[P3] : 필요한 어노테이션만 붙히는게 좋아보입니다!

AoiTuNa replied:
검증하고 필요한 어노테이션만 추가했습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/domain/UserLoginRequestDto.java:
```diff
@@ -0,0 +1,17 @@
+ * @version 1.0
+ */
+@Getter
+@Setter
```
[P3] : 여기도 필요한 어노테이션만 붙히는게 좋을 것 같습니다

AoiTuNa replied:
검증하고 필요한 어노테이션만 추가했습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/service/impl/AuthServiceImpl.java:
```diff
@@ -0,0 +1,35 @@
+        try {
+            authAdapter.logout(token);
+        }catch (Exception e){
+            e.printStackTrace();
```
[P4] : 로그아웃 실패 시 에러 핸들링이 되어야 할 것 같습니다

AoiTuNa replied:
HttpClientErrorException으로 handling 했습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/adapter/impl/AuthAdapterImpl.java:
```diff
@@ -0,0 +1,67 @@
+        requestEntity,
+        Void.class
+    );
+    if(exchange.getStatusCode() != HttpStatus.OK){
```
[P3] : RestTemplate에서 보낸 요청의 응답이 400, 500 번대면 62번째 라인이 실행 되지 않고 exception이 발생합니다. 한 번 테스트 해보시고 resttemplate을 빈에 등록할 때 커스텀한 에러 핸들러를 사용해서 직접 에러 핸들링을 사용할 수 있습니다
https://www.baeldung.com/spring-rest-template-error-handling

AoiTuNa replied:
해당부분을 삭제하고 보내는 400 , 500 번대 오류를 서비스 단에서 처리하도록 수정했습니다!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/advice/HttpExceptions.java:
```diff
@@ -0,0 +1,45 @@
+      Cookie[] cookies = req.getCookies();
+      if(Arrays.stream(cookies).filter(cookie -> "authorization".equals(cookie.getName()))
+          .findFirst().isEmpty()){
+        redirectAttributes.addFlashAttribute("error","don't have authorization token");
```
[P4] : 사용자가 보게 될 메시지는 한글이 더 명확할 것 같습니다. 아니면 배웠었던 locale을 활용해보는 것도..

AoiTuNa replied:
일단은 한글로 바꾸어 놓았습니다! 차후 html단에서 메시지 출력에 사용할 부분만 한글로 추가해 볼게요!

ckddms6530 commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+ * @see #doLogout(HttpServletRequest, RedirectAttributes)
+ */
+@Controller
+@RequestMapping()
```
[P3] : 불필요한 어노테이션인 것 같습니다

AoiTuNa replied:
제거 했습니다!

ckddms6530 reviewed (commented):
페이지 구현하면서 필요했던 리소스들만 업로드하면 좋을 것 같습니다

AoiTuNa committed:
코드리뷰
HttpServletRequest 를 사용해서 쿠키 값을 가져옴
-> @CookieValue 어노테이션을 활용

로그아웃 로직의 토큰이 없을때 예외 수정
403 ->401

AoiTuNa committed:
불필요한 else 문 제거

AoiTuNa committed:
domain 불필요한 어노테이션 제거

chanhwiim commented:
만약 리뷰가 왔고, 의견의 정리가 끝났거나, 코드의 수정이 끝났다면, Resolve conversation 해주세요

chanhwiim commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+     * 추가적인 처리를 위한 authService
+     * @see com.nhnacademy.front.server.service.AuthService
+     */
+    private final AuthService authService;
```
p4: javadoc은 private에는 쓰지 않는다고 알고 있습니다.

AoiTuNa replied:
제거 했습니다

chanhwiim commented on src/main/java/com/nhnacademy/front/server/controller/AuthController.java:
```diff
@@ -0,0 +1,102 @@
+    public String showLoginForm(){return "pages/auth/login";}
+
+
+    //Id -> email 상황따라 변경
```
p3: 일단 id로 하기로 결정이 된 걸로 기억합니다. Id로 바꿔주세욧

AoiTuNa replied:
수정 했습니다

AoiTuNa committed:
authAdapter 부분의 statusCode부분의 코드가 실행되지 않는 문제를
실제로 httpClientException을 받아서 advice로 처리한 다음 로그인으로 리다이렉트 하게 처

AoiTuNa committed:
테스트 로직 추가 및 에러메시지 한글로 수정

AoiTuNa committed:
javadoc private 부분 제

AoiTuNa committed:
email -> id 변경

AoiTuNa committed:
html수정

AoiTuNa committed:
테스트 코드 수정

AoiTuNa committed:
Todo 삭제 및 gateWay 절대경로 설정

AoiTuNa committed:
안쓰는 라이브러리 삭제

AoiTuNa committed:
gateway 실제 주소 적용하고 테스트 코드 적용을 안했네요!
